src/main/kotlin/com/ampnet/identityservice/blockchain/BlockchainServiceImpl.kt

```diff
@@ -4,40 +4,47 @@ import com.ampnet.identityservice.config.ApplicationProperties
 import com.ampnet.identityservice.service.unwrap
 import mu.KotlinLogging
 import org.springframework.stereotype.Service
+import org.web3j.abi.FunctionEncoder
+import org.web3j.abi.datatypes.Address
+import org.web3j.abi.datatypes.Function
 import org.web3j.crypto.Credentials
+import org.web3j.crypto.RawTransaction
 import org.web3j.protocol.Web3j
 import org.web3j.protocol.core.DefaultBlockParameterName
 import org.web3j.protocol.core.RemoteFunctionCall
 import org.web3j.protocol.core.Request
 import org.web3j.protocol.core.Response
-import org.web3j.protocol.core.methods.request.Transaction
 import org.web3j.protocol.http.HttpService
+import org.web3j.tx.RawTransactionManager
 import org.web3j.tx.ReadonlyTransactionManager
 import org.web3j.tx.gas.DefaultGasProvider
 import java.io.IOException
+import java.math.BigInteger
 
 private val logger = KotlinLogging.logger {}
 
 @Service
 class BlockchainServiceImpl(private val applicationProperties: ApplicationProperties) : BlockchainService {
 
-    private val web3j by lazy { Web3j.build(HttpService(applicationProperties.provider.blockchainApi)) }
+    private val web3j: Web3j by lazy { Web3j.build(HttpService(applicationProperties.provider.blockchainApi)) }
+    private val credentials: Credentials by lazy { Credentials.create(applicationProperties.smartContract.privateKey) }
 
-    @Suppress("ReturnCount")
+    @Suppress("ReturnCount", "MagicNumber")
     override fun whitelistAddress(address: String, issuerAddress: String): String? {
         logger.info { "Whitelisting address: $address" }
-        val credentials = Credentials.create(applicationProperties.smartContract.privateKey)
-        val contract = IIssuer.load(issuerAddress, web3j, credentials, DefaultGasProvider())
-        val approveWalletCall = contract.approveWallet(address).encodeFunctionCall()
-        val nonce = web3j.ethGetTransactionCount(address, DefaultBlockParameterName.LATEST)
-            .sendSafely()?.transactionCount ?: return null
-        val gasPrice = web3j.ethGasPrice().sendSafely()?.gasPrice ?: return null
-        val transaction = Transaction
-            .createContractTransaction(credentials.address, nonce, gasPrice, approveWalletCall)
-
-        val transactionHash = web3j.ethSendTransaction(transaction).sendSafely()?.transactionHash
-        logger.info { "Sent transaction: $transactionHash" }
-        return transactionHash
+        val nonce = web3j.ethGetTransactionCount(credentials.address, DefaultBlockParameterName.LATEST)
+            .send().transactionCount ?: return null
+        val gasPrice = web3j.ethGasPrice().send().gasPrice ?: return null
+        val function = Function("approveWallet", listOf(issuerAddress.toAddress(), address.toAddress()), emptyList())
```

Comment:
you can replace string with IIssuer.FUNC_APPROVEWALLET

Reply:
This is not call on Issuer contract.